src/Chapter2.hs

```diff
@@ -857,7 +886,12 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind lst = rewind' lst []
+
+rewind' :: [a] -> [a] -> [a]
+rewind' [] acc = acc
+rewind' (x:xs) acc = rewind' xs (x:acc)
```

Comment:
Usually, instead of creating a separate top-level function and taking names from the global namespace, it's recommended to create local functions inside the `where` clause. In most cases, this function is called `go` and it's known as the "Recursive go" pattern.

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go